lib/utils.py

```diff
@@ -46,3 +53,35 @@ def is_in_uk(simple_polygons):
         first_coordinate[1] > uk_south_west[1] and
         first_coordinate[1] < uk_north_east[1]
     )
+
+
+def upload_to_s3(config, rendered_pages):
+    session = boto3.session.Session(
+        aws_access_key_id=config["BROADCASTS_AWS_ACCESS_KEY_ID"],
+        aws_secret_access_key=config["BROADCASTS_AWS_SECRET_ACCESS_KEY"],
+        region_name=config["BROADCASTS_AWS_REGION"],
+    )
+
+    s3 = session.resource('s3')
+    bucket_name = config['GOVUK_ALERTS_BUCKET_NAME']
+
+    for path, content in rendered_pages.items():
+        logger.info("Uploading " + path)
+        item = s3.Object(bucket_name, path)
+        item.put(Body=content, ContentType="text/html")
+
+
+def purge_cache(config):
+    fastly_service_id = config['FASTLY_SERVICE_ID']
+    fastly_api_key = config['FASTLY_API_KEY']
+    surrogate_key = config['FASTLY_SURROGATE_KEY']
+    fastly_url = f"https://api.fastly.com/{fastly_service_id}/purge/{surrogate_key}"
```

Comment:
We're missing a `service` here: `...api.fastly.com/service/{fastly_service_id}...`.

Reply:
Fixed in https://github.com/alphagov/notifications-govuk-alerts/pull/159/commits/8d46dfedeb69d282354878eb0c81a1671a547f05